rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -685,6 +683,16 @@ impl Controller {
                 })?;
                 Ok(())
             }
+            IpcServerMsg::ConnectResult(result) => match result {
+                Ok(()) => Ok(ran_before::set().await?),
```

Comment:
If the first run of Firezone hits a connection error, we still show the Welcome screen on the next run